### PR TITLE
Refactoring and fix issues

### PR DESCRIPTION
**Pull request by thebluber**
I'd like to propose a way to use our own response classes instead of OpenStruct.

### CODE REVIEW

File: lib/cirro_io_v2/request_clients/jwt.rb [hunk 1]
```diff
@@ -14,7 +14,6 @@ def initialize(base_url:, private_key:, client_id:)
         @connection = Faraday.new(url: base_url) do |conn|
           conn.request :json
           conn.response :json
-          conn.adapter :net_http
```
Comment (thebluber) [inline lib/cirro_io_v2/request_clients/jwt.rb:17]: @aliaksandr-volkau which routes were you calling? I didn't get any errors without this line 🤷‍♀️

File: lib/cirro_io_v2/responses/base.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            data: body[:data].map { |list_item| list_item_class.new(list_item) }
+          )
+          super(*self.members.map { |attr| values_hash[attr] })
+        elsif has_list?(body)
+          super(
+            *body.slice(*members.excluding(has_list?(body))).values,
+            create_sub_resource(has_list?(body), body[has_list?(body)])
+          )
+        else
+          super(*body.slice(*members).values)
```
Comment (garciaf) [inline lib/cirro_io_v2/responses/base.rb:19]: what is `members` ?

File: spec/cirro_io_v2/resources/gig_spec.rb [hunk 3]
```diff
@@ -14,8 +14,8 @@
     end
 
     it 'returns gig structure' do
-      expect(create_gig).to be_a(Struct)
-      expect(create_gig.to_h).to eq(gig.to_h)
+      expect(create_gig).to be_a(CirroIOV2::Responses::GigResponse)
```
Comment (garciaf) [inline spec/cirro_io_v2/resources/gig_spec.rb:17]: Is there a test where you check the behavior when you receive a list of gig-invitations?

https://developers.cirro.io/api-docs/v1#get-a-list-of-gig-invitations

I would suggest to check this use case as well.

File: lib/cirro_io_v2/responses/base.rb (hunk 2 above)
Comment (aliaksandr-volkau) [reply lib/cirro_io_v2/responses/base.rb:19]: It is a list of attributes of Struct. For example: struct = Struct.new(:foo, :bar). struct.members => [:foo, :bar]

File: lib/cirro_io_v2/responses/responses.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
+module CirroIOV2
+  module Responses
```
Comment (garciaf) [inline lib/cirro_io_v2/responses/responses.rb:2]: I am confused by the usage of a separate class as a response. 
Why the resource itself can not be the response itself?

What constraint or reason we need to go through the response class

**Review by garciaf — APPROVED** [review]

**Review by mesmerze — APPROVED** [review]
🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥

**Pull request merged**